components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1461,9 +1486,9 @@ where
 
     // Returns `None` if the `msg` doesn't contain a snapshot or it contains a snapshot which
     // doesn't conflict with any other snapshots or regions. Otherwise a `SnapKey` is returned.
-    fn check_snapshot(&mut self, msg: &RaftMessage) -> Result<Option<SnapKey>> {
+    fn check_snapshot(&mut self, msg: &RaftMessage) -> Result<Either<SnapKey, Vec<(u64, bool)>>> {
         if !msg.get_message().has_snapshot() {
```

Comment:
The check can be removed because it's only called when the message is snapshot.

Reply:
That's not true..